qiskit_nature/algorithms/ground_state_solvers/adapt_vqe.py

```diff
@@ -89,22 +104,24 @@ def _compute_gradients(
 
         Returns:
             List of pairs consisting of gradient and excitation operator.
+            parameters = ParameterVector("θ", self._num_parameters)
+            times = ParameterVector("t", self.reps * len(self.operators))
         """
         res = []
-        # compute gradients for all excitation in operator pool
         for exc in self._excitation_pool:
-            # add next excitation to ansatz
             self._ansatz.operators = self._excitation_list + [exc]
-            # set the current ansatz
-            vqe.ansatz = self._ansatz
-            # evaluate energies
-            parameter_sets = theta + [-self._delta] + theta + [self._delta]
-            energy_evaluation = vqe.get_energy_evaluation(self._main_operator)
-            energy_results = energy_evaluation(np.asarray(parameter_sets))
-            # compute gradient
-            gradient = (energy_results[0] - energy_results[1]) / (2 * self._delta)
-            res.append((np.abs(gradient), exc))
-
+            if self.gradient._grad_method.analytic:
```

Comment:
```suggestion
            if self.gradient.grad_method.analytic:
```
`grad_method` is also available as a property so we should prefer accessing it via this public attribute.